### PR TITLE
Add prometheus target to argocd-metrics service

### PR DESCRIPTION
**Pull request by kubasobon**
Towards https://github.com/giantswarm/giantswarm/issues/17444

### CODE REVIEW

**Comment by kubasobon** [comment]
@JosephSalisbury you mentioned an alternative for label/annotation-based service discovery. What was it?

**Comment by JosephSalisbury** [comment]
[ServiceMonitors](https://intranet.giantswarm.io/docs/product/architecture-specs-adrs/specs/configuration-of-targets-in-tc-prometheus/#custom-monitors)

**Comment by JosephSalisbury** [comment]
But if the service label/annotation approach works, then that's legittt

**Review by JosephSalisbury — APPROVED** [review]
lgtmiiwfy

File: manifests/argocd/metrics.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+apiVersion: v1
+kind: Service
+metadata:
+  annotations:
+    giantswarm.io/monitoring-port: "8082"
+  labels:
+    giantswarm.io/monitoring: "true"
+  name: argocd-metrics
```
Comment (kopiczko) [inline manifests/argocd/metrics.yaml:8]: How about renaming to `service-metrics-patch.yaml`. I was super confused this is a full service definition at first.

**Review by kopiczko — APPROVED** [review]

**Pull request merged**